packages/appium/lib/appium.js

```diff
@@ -662,14 +655,16 @@ class AppiumDriver extends DriverCore {
 
   /**
    * Creates instances of all of the enabled Plugin classes
+   * @param {string|null} driverId - ID to use for linking a driver to a plugin in logs
    * @returns {Plugin[]}
    */
-  createPluginInstances() {
+  createPluginInstances(driverId = null) {
     /** @type {Plugin[]} */
     const pluginInstances = [];
     for (const [PluginClass, name] of this.pluginClasses.entries()) {
       const cliArgs = this.getCliArgsForPlugin(name);
-      const plugin = new PluginClass(name, cliArgs);
+      const plugin = new PluginClass(name, cliArgs, driverId);
+      /** @type {Plugin & ExtensionCore} */(plugin).updateBidiCommands(PluginClass.newBidiCommands ?? {});
```

Comment:
this would probably fail if an older base plugin is used. Need to explicitly check if updateBidiCommands fmethod is present on the instance

Reply:
I don't think this would fail. You can send extra parameters to functions in JS if you want, without error.

Reply:
oh, you're referring to the `updateBidiCommands`, yes you're right.

Reply:
It's not about parameters, but rather about the presence of `updateBidiCommands` instance method